src/3rdparty/walkontable/src/table.js

```diff
@@ -730,129 +730,199 @@ class Table {
+   *           +--------------+                                  ---+ last rendered row
+   *                                                                │
+   *                                                                │ TRUE
    *
    * @param {nunber} row The visual row index.
    * @returns {boolean}
    */
+  /* eslint-enable jsdoc/require-description-complete-sentence */
   isRowAfterRenderedRows(row) {
-    if (row < 0) {
-      const columnHeaders = this.wot.getSetting('columnHeaders');
-      const columnHeadersCount = columnHeaders.length;
-      const zeroBasedHeaderLevel = columnHeadersCount + row;
-      return this.isColumnHeaderLevelRendered(zeroBasedHeaderLevel);
-    }
     return row > this.getLastRenderedRow();
   }
 
-  isColumnBeforeViewport(column) {
-    return this.columnFilter && (this.columnFilter.sourceToRendered(column) < 0 && column >= 0);
-  }
-
+  /* eslint-disable jsdoc/require-description-complete-sentence */
   /**
-   * Check if the given column index is smaller than the index of the first column that is currently redered
-   * and return TRUE in that case, or FALSE otherwise.
+   * Check if the given column index is smaller than the index of the first column that
+   * is currently rendered and return TRUE in that case, or FALSE otherwise.
+   *
+   * Negative column index is used to check the rows' headers.
    *
-   * Negative column index is used to check the header cells. As a simplification, it checks negative column index
-   * the same way as a regular column 0. You can interpret this as follows: If the column 0 is rendered, all header
-   * cells are also rendered.
+   *                            For fixedColumnsLeft: 1 the master overlay
+   *                            do not render this first columns.
+   *  Headers    -3   -2   -1    |
+   *           +----+----+----║┄ ┄ +------+------+
+   *           │    │    │    ║    │  B1  │  C1  │
+   *           +--------------║┄ ┄ --------------│
+   *           │    │    │    ║    │  B2  │  C2  │
+   *           +--------------║┄ ┄ --------------│
+   *           │    │    │    ║    │  B3  │  C3  │
+   *           +----+----+----║┄ ┄ +------+------+
+   *                               ╷             ╷
+   *      -------------------------+-------------+---------------->
+   *          TRUE             first    FALSE   last         FALSE
+   *                           rendered         rendered
+   *                           column           column
    *
    * @param {number} column The visual column index.
    * @returns {boolean}
    */
+  /* eslint-enable jsdoc/require-description-complete-sentence */
   isColumnBeforeRenderedColumns(column) {
     const first = this.getFirstRenderedColumn();
 
-    if (column < 0) {
-      column = 0;
+    // Check the headers only in case when the first rendered column is -1 or 0.
+    // This is an indication that the overlay is placed on the most top-left position.
```

Comment:
`is placed on the most top-left position.`

Not just on the left one?

Reply:
Fixed within https://github.com/handsontable/handsontable/pull/7006/commits/4d711c8bd24bcc82016fb6327d947e60851aed64.